File: MinElemLinkedList/Find.go
package main;
import "fmt"

type Node struct {
	data int;
	next * Node;
}

type NODE = Node;
type PNODE = * Node;
type PPNODE = ** Node;

func InsertFirst(Head PPNODE,value int) {
	var newn PNODE = new(NODE);
	newn.data = value;
	newn.next = nil;
	
	if(*Head == nil) {
		*Head = newn;
	} else {
		newn.next = *Head;
		*Head = newn;
	}
}

func InsertLast(Head PPNODE,value int) {
	var newn PNODE = new(NODE);
	newn.data = value;
	newn.next = nil;

	var temp PNODE = *Head;

	if(*Head == nil) {
		*Head = newn;
	} else {
		for(temp.next != nil) {
			temp = temp.next;
		}
		temp.next = newn;
	}
}

func Display(Head PNODE) {
	for(Head != nil) {
		fmt.Printf("|%d|->",Head.data);
		Head = Head.next;
	}
}

func Count(Head PNODE) int {
	var icnt int = 0;

	for(Head != nil) {
		icnt++;
		Head = Head.next;
	}

	return icnt;
}

func SmallestElem(Head PNODE) int {
	var Min int = 0;

	Min = Head.data;
	for(Head != nil) {
		if(Head.data < Min) {
			Min = Head.data;
		}
		Head = Head.next;
	}

	return Min;
}

func main() {
	var iret int = 0;

	var First PNODE = nil;

	fmt.Println("Jay Ganesh....");

	InsertFirst(&First,51);
	InsertFirst(&First,11);
	InsertFirst(&First,10);
	InsertLast(&First,2);
	InsertLast(&First,101);
	InsertLast(&First,120);

	fmt.Println("Elements of Linked list are : ");
	Display(First);

	iret = Count(First);
	fmt.Printf("\nCount of Elements From Linked List is : %d\n",iret);

	iret = SmallestElem(First);
	fmt.Printf("Smallest Element From Linked List is : %d\n",iret);	
}

File: MinElemLinkedList/Find.cpp
#include<iostream>
using namespace std;

struct Node 
{
    int data;
    struct Node *next;
};

typedef struct Node NODE;
typedef struct Node * PNODE;
typedef struct Node ** PPNODE;

class SinglyLL
{
    private : 
        PPNODE Head;
    public : 
        SinglyLL();
        void InsertFirst(PPNODE,int);
        void InsertLast(PPNODE,int);
        void Display(PNODE);
        int Count(PNODE);
        int SmallElem(PNODE);
};

SinglyLL :: SinglyLL()
{
    Head = NULL;
}
inline void SinglyLL::InsertFirst(PPNODE Head,int value)
{
    PNODE newn = (PNODE)malloc(sizeof(NODE));
    newn->data = value;
    newn->next = NULL;

    if(*Head == NULL)
    {
        *Head = newn;
    }
    else
    {
        newn->next = *Head;
        *Head = newn;
    }
}

inline void SinglyLL::InsertLast(PPNODE Head,int value)
{
    PNODE newn = (PNODE)malloc(sizeof(NODE));
    newn->data = value;
    newn->next = NULL;
    PNODE temp = *Head;

    if(*Head == NULL)
    {
        *Head = newn;
    }
    else 
    {
        while(temp->next != NULL)
        {
            temp = temp->next;
        }
        temp->next = newn;
    }
}

inline void SinglyLL::Display(PNODE Head)
{
    while(Head != NULL)
    {
        cout<<"|"<<Head->data<<"|->";
        Head = Head->next;
    }
}

inline int SinglyLL::Count(PNODE Head)
{
    int icnt = 0;

    while(Head != NULL)
    {
        icnt++;
        Head = Head->next;
    }

    return icnt;
}

inline int SinglyLL::SmallElem(PNODE Head)
{
    int Min = 0;

    Min = Head->data;
    while(Head != NULL)
    {
        if(Head->data < Min)
        {
            Min = Head->data;
        }
        Head = Head->next;
    }

    return Min;
}

int main()
{
    PNODE First = NULL;
    int iret = 0;

    cout<<"Jay Ganesh....\n";

    SinglyLL *sobj = new SinglyLL();
    sobj->InsertFirst(&First,51);
    sobj->InsertFirst(&First,11);
    sobj->InsertFirst(&First,10);
    sobj->InsertLast(&First,2);
    sobj->InsertLast(&First,101);
    sobj->InsertLast(&First,105);

    cout<<"Elements of linked List is : \n";
    sobj->Display(First);

    iret = sobj->Count(First);
    cout<<"\nCount of Elements From Linked List Is : "<<iret<<"\n";

    iret = sobj->SmallElem(First);
    cout<<"Smallest Element From Linked List Is : "<<iret<<"\n";
    
    return 0;
}